src/file_pipe_log/pipe_builder.rs

```diff
@@ -528,13 +528,14 @@ impl<F: FileSystem> DualPipesBuilder<F> {
 
     fn initialize_files(&mut self) -> Result<()> {
         let target_file_size = self.cfg.target_file_size.0 as usize;
-        let mut target = std::cmp::min(
-            self.cfg.prefill_capacity(),
-            self.cfg
-                .recycle_capacity()
-                .saturating_sub(self.append_files.len()),
-        );
-        let to_create = target.saturating_sub(self.recycled_files.len());
+        let prefill_target =
+            std::cmp::min(self.cfg.prefill_capacity(), self.cfg.recycle_capacity())
```

Comment:
> the old logic is that prefill_capacity is the number of recycle files that should be pre-created even if there are already some append files; but the new logic changes the semantic to let prefill_capacity the total files number (recycle + append), I think this is more intuitive.

This doesn't seem more intuitive to me. The `prefill_limit` stands for the amount of data that user can tolerate to write before startup. It should not be subtracted with append files.

Reply:
ok. I reverted this change as the default `purge_threshold` and `recycle_threshold` is still the same.